war/src/main/java/io/lumeer/engine/rest/CollectionService.java

```diff
@@ -284,19 +289,14 @@ public void dropAttribute(final @PathParam("collectionName") String collectionNa
     *       When the given collection does not exist.
     * @throws UnauthorizedAccessException
     *       When current user is not allowed to write to the collection
-    * @throws CollectionMetadataDocumentNotFoundException
-    *       when metadata for collection was not found
     */
    @POST
    @Path("/{collectionName}/meta/{attributeName}")
-   @Consumes(MediaType.APPLICATION_JSON)
-   public void addCollectionMetadata(final @PathParam("collectionName") String collectionName, final @PathParam("attributeName") String attributeName, final DataDocument metadata) throws CollectionNotFoundException, UnauthorizedAccessException, CollectionMetadataDocumentNotFoundException {
+   public void addCollectionMetadata(final @PathParam("collectionName") String collectionName, final @PathParam("attributeName") String attributeName, final DataDocument metadata) throws CollectionNotFoundException, UnauthorizedAccessException {
       if (collectionName == null || attributeName == null) {
          throw new BadRequestException();
       }
-      if (!checkCollectionForWrite(getInternalName(collectionName))) {
-         throw new UnauthorizedAccessException();
-      }
+      // TODO: What about access rights?
```

Comment:
In order to modify attributes, a user needs to have `write` role on a given collection.